lib/aggregate/attribute/string.rb

```diff
@@ -1,17 +1,54 @@
 class Aggregate::Attribute::String < Aggregate::Attribute::Builtin
 
+  class Aggregate::EncryptionError < StandardError; end;
+
   def self.available_options
     Aggregate::Attribute::Builtin.available_options + [
       :size, # The maximum length of the string
+      :encrypted, # Whether or not we need to encrypt a field
     ]
   end
 
   def load(value)
-    value.to_s
+    options[:encrypted] ? decrypt(value) : value.to_s
+  end
+
+  def decrypt(value)
+    # get string, convert to hash, read decoded iv, decode value, decrypt with iv and value
+    hash = ActiveSupport::JSON.decode value
+    Aggregate::Base.secret_keys_from_config.empty? and raise Aggregate::EncryptionError, "must specify a key for decryption"
+
+    find_decrypted_value(Base64.strict_decode64(hash["encrypted_data"]), Base64.strict_decode64(hash["initilization_vector"]))
+  end
+
+  def find_decrypted_value(value, iv)
+    encrypted_value = nil
+    Aggregate::Base.secret_keys_from_config.find do |key|
+      begin
+        encrypted_value = Encryptor.decrypt(value: value, key: key, iv: iv)
+      rescue OpenSSL::Cipher::CipherError
+        nil
+      end
+    end
+    encrypted_value.presence or raise Aggregate::EncryptionError, "could not decrypt #{name} because the correct decryption key is not found"
+    encrypted_value
```

Comment:
`#presence` should return the value of `encrypted_value` so you don't need to do that here.